src/cache/disk.rs

```diff
@@ -38,12 +80,20 @@ impl DiskCache {
     pub fn new<T: AsRef<OsStr>>(root: &T, max_size: u64, pool: &ThreadPool) -> DiskCache {
         DiskCache {
             //TODO: change this function to return a Result
-            lru: Arc::new(Mutex::new(
-                LruDiskCache::new(root, max_size).expect("Couldn't instantiate disk cache!"),
-            )),
+            lru_init: Mutex::new(DiskCacheInit::new(root.as_ref(), max_size)),
             pool: pool.clone(),
         }
     }
+
+    fn lru(&self) -> Arc<Mutex<LruDiskCache>> {
+        let mut lru_init = self.lru_init.lock().unwrap();
+
+        if let Some(lru) = lru_init.opt_lru() {
```

Comment:
This fast path is needed because of the `mem::replace` in the slow path. Might be able to elide it with a `BackgroundLazyInit` mechanism.

Reply:
Perhaps a dumb question, but why not simply add the fast path at the start of DiskCacheInit's lru function?
```rust
if let Self::Initialized(lru) = self {
   return lru;
}
```

Reply:
Indeed, thanks. :)